plugins/iamauth.js

```diff
@@ -175,8 +187,10 @@ class IAMPlugin extends BasePlugin {
               debug('Successfully renewed IAM session.');
               callback();
             } else {
-              self.shouldApplyIAMAuth = false;
-              callback(new Error('Failed to exchange IAM token with Cloudant'));
+              if (response.statusCode < 500) {
+                self.shouldApplyIAMAuth = false; // no retry
```

Comment:
oh since we'd need to log here as well maybe logging where the callback is consumed is better